src/aead/block.rs

```diff
@@ -78,14 +79,56 @@ impl Block {
 impl From<&'_ [u8; BLOCK_LEN]> for Block {
     #[inline]
     fn from(bytes: &[u8; BLOCK_LEN]) -> Self {
-        unsafe { core::mem::transmute_copy(bytes) }
+        Self {
+            subblocks: [
+                u64::from_le_bytes(
+                    bytes[..8]
+                        .try_into()
+                        .expect("error converting leading bytes"),
```

Comment:
Use `expect` here and other places, because the conversion (`[u8] -> [u8; 8]`) should never fail. If conversion does fail, panicking is IMHO the correct behavior, rather than silently using such corrupted data or default values (i.e. `[0u8; 8]`).

Reply:
From the [TryFrom](https://doc.rust-lang.org/nightly/src/core/array/mod.rs.html#157) docs:

```
    fn try_from(slice: &[T]) -> Result<&[T; N], TryFromSliceError> {
        if slice.len() == N {
            let ptr = slice.as_ptr() as *const [T; N];
            // SAFETY: ok because we just checked that the length fits
            unsafe { Ok(&*ptr) }
        } else {
            Err(TryFromSliceError(()))
        }
    }
```

So, it only fails when the lengths don't match up. Here, they are guaranteed to match.

Reply:
From the [TryFrom](https://doc.rust-lang.org/src/core/array/mod.rs.html#157) docs:

```
    fn try_from(slice: &[T]) -> Result<&[T; N], TryFromSliceError> {
        if slice.len() == N {
            let ptr = slice.as_ptr() as *const [T; N];
            // SAFETY: ok because we just checked that the length fits
            unsafe { Ok(&*ptr) }
        } else {
            Err(TryFromSliceError(()))
        }
    }
```

So, it only fails when the lengths don't match up. Here, they are guaranteed to match.

Reply:
Please just use `unwrap()`. I don't want to have the strings passed to `expect()` included in the resulting library.

Reply:
> Please just use unwrap()

Sounds good. Would comments about the safety of using `unwrap()` here also be good (to the effect of "lengths guaranteed to match, see the docs...")?